routes/artifacts.js

```diff
@@ -163,13 +163,23 @@ api.declare({
 
   // Ensure that the run is running
   if (run.state !== 'running') {
-    return res.status(409).json({
-      message:    "The given run is not running",
-      error: {
-        status:   task.status(),
-        runId:    runId
-      }
-    });
+    var allow = false;
+    if (run.state === 'exception') {
+      // If task was resolved exception, we'll allow artifacts to be uploaded
+      // up to 25 min past resolution. This allows us to report exception as
+      // soon as we know and then upload logs if possible.
+      // Useful because exception usually implies something badly wrong.
+      allow = (new Date(run.resolved).getTime() > Date.now() - 25 * 60 * 1000);
+    }
+    if (!allow) {
+      return res.status(409).json({
+        message:    "The given run is not running",
```

Comment:
Well, it's a bit more complicated, because resolved as `completed` won't allow for upload 25min past resolution.

But yeah, making it a lot more explanatory can't hurt. Certainly not, I'll push a better messages. Thanks for pointing this out, we probably have a lot of error messages that could be more explanatory...